tensorflow_addons/optimizers/lars_optimizer.py

```diff
@@ -0,0 +1,190 @@
+        'bias']
+      use_nesterov: when set to True, nesterov momentum will be enabled
+
+    Raises:
+      ValueError: If a hyperparameter is set to a non-sensical value.
+    """
+    if momentum < 0.0:
+      raise ValueError("momentum should be positive: %s" % momentum)
+    if weight_decay < 0.0:
+      raise ValueError("weight_decay should be positive: %s" % weight_decay)
+    super(LARSOptimizer, self).__init__(name=name, **kwargs)
+
+    self._skip_list = skip_list
+    self._use_nesterov = use_nesterov
+    self._set_hyper('learning_rate', learning_rate)
+    self._set_hyper('momentum', momentum)
+    self._set_hyper('weight_decay', weight_decay)
+    self._set_hyper('eeta', eeta)
+    self._set_hyper('epsilon', epsilon)
+    self._set_hyper('use_nesterov', use_nesterov)
+
+  def get_config(self):
+    config = {
+        'learning_rate': self._serialize_hyperparameter('learning_rate'),
+        'momentum': self._serialize_hyperparameter('momentum'),
+        'weight_decay': self._serialize_hyperparameter('weight_decay'),
+        'eeta': self._serialize_hyperparameter('eeta'),
+        'epsilon': self._serialize_hyperparameter('epsilon'),
+        'use_nesterov': self._serialize_hyperparameter('use_nesterov')
+    }
+    base_config = super(LARSOptimizer, self).get_config()
+    return dict(list(base_config.items()) + list(config.items()))
+
+  def _create_slots(self, var_list):
+    for var in var_list:
+      self.add_slot(var, 'momentum')
+
+  def compute_lr(self, grad, var):
+    scaled_lr = self._serialize_hyperparameter('learning_rate')
+    if self._skip_list is None or not any(v in var.name
+                                          for v in self._skip_list):
+      w_norm = tf.linalg.norm(var, ord=2)
+      g_norm = tf.linalg.norm(grad, ord=2)
+      eeta = self._serialize_hyperparameter('eeta')
+      weight_decay = self._serialize_hyperparameter('weight_decay')
+      epsilon = self._serialize_hyperparameter('epsilon')
+      trust_ratio = array_ops.where_v2(
+          math_ops.greater(w_norm, 0),
+          array_ops.where_v2(
+              math_ops.greater(g_norm, 0),
+              (eeta * w_norm /
+               (g_norm + weight_decay * w_norm + epsilon)), 1.0),
+          1.0)
+      scaled_lr = self._serialize_hyperparameter('learning_rate') * trust_ratio
+      # Add the weight regularization gradient
+      grad = grad + weight_decay * var
+    return scaled_lr, grad
+
+  def _apply_dense(self, grad, var):
+    scaled_lr, grad = self.compute_lr(grad, var)
+    mom = self.get_slot(var, "momentum")
+    momentum = self._serialize_hyperparameter('momentum')
+    use_nesterov = bool(self._serialize_hyperparameter('use_nesterov'))
+    return training_ops.apply_momentum(
+        var,
+        mom,
+        math_ops.cast(1.0, var.dtype.base_dtype),
+        grad * scaled_lr,
+        momentum,
+        use_locking=False,
+        use_nesterov=use_nesterov)
```

Comment:
I don't think this function is needed according to [this comment](https://github.com/tensorflow/tensorflow/blob/1cf0898dd4331baf93fe77205550f2c2e6c90ee5/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L219)